### DISCUSSION TITLE
Go to surrounding function's declaration

### DISCUSSION THREAD

**Discussion by jrvidal** [helix-editor/helix#6500] [Q&A]
I find myself wanting to jump to the declaration/name of the surrounding function often. I just learned about `maf`, but it is not quite the same, as I might need to `A-;` to the beginning of the selection if the body is large.

Does this exist? And if not, would it be worthy of adding to core?

**Comment by pascalkuthe** [comment]
I have been thinking that we might want to change the position of the cursor at the start of the selection rather than at the end of the selection. For example recently we changed the pickerto highlight the selection instead of placing the cursor at the first line of the picker item. This leads to somewhat disorienting behavior when doing `gd` on modules in rust for example. I think actually placing the cursor in the cursor might be a better default, what do you think @the-mikedavis

**Reply by the-mikedavis** [reply]
For `]f`/`[f` I think the direction of the selection is important but I don't have strong feelings about `maf`/`mif`. Backwards selections (cursor at the beginning) would probably be more intuitive 👍

**Reply by pascalkuthe** [reply]
Yeah for the unimpaired commands totally I agree I meant stuff like the picker/`gd` and `maf`/`mif` that's not really directional

**Reply by jrvidal** [reply]
That would help, but `maf` is kind of a "hack", in the sense that my original intent is to get to the function _declaration_ (i.e. its name identifier), and not necessarily select anything.

But if nothing else, I would totally take it :+1:

**Reply by rkshthrmsh** [reply]
A command like `goto_function_start` could be useful. Unsure how this would need to be implemented, but it might be analogous to `goto_line_start` for lines.

**Reply by willhansen** [reply]
binding `gf` to `goto_function_start` could be good
